DOC: add note to docstring about `dtype` keyword when creating Series from abother Series

taranarmo opened the pull request:
When creating Series from another Series `dtype` keyword is not not used. Add note on `dtype` keyword to Series and DataFrame constructor. See GH #59060.

- [x] closes #59060 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

mroeschke reviewed (changes requested):
A warning would be too noisy. A note in the docs would be sufficient.

taranarmo committed:
BUG: add note on creating Series from Series with dtype keyword

When creating Series from another Series `dtype` keyword is ignored.
See GH #59060. Add note to the docstring to underline this behaviour to
Series and DataFrame constructors.

taranarmo commented:
Undo the previous changes and add a note to the `dtype` keyword on `pd.Series` and `pd.DataFrame` that says that the keyword is ignored if `data` is Series and DataFrame correspondingly.

mroeschke commented on pandas/core/frame.py:
```diff
@@ -531,6 +531,7 @@ class DataFrame(NDFrame, OpsMixin):
         will perform column selection instead.
     dtype : dtype, default None
         Data type to force. Only a single dtype is allowed. If None, infer.
+        If `data` is DataFrame then is ignored.
```
```suggestion
        If ``data`` is DataFrame then is ignored.
```

taranarmo replied:
Why do we need double backticks if elsewhere they are single?

mroeschke commented on pandas/core/series.py:
```diff
@@ -256,6 +256,7 @@ class Series(base.IndexOpsMixin, NDFrame):  # type: ignore[misc]
         Data type for the output Series. If not specified, this will be
         inferred from `data`.
         See the :ref:`user guide <basics.dtypes>` for more usages.
+        If `data` is Series then is ignored.
```
```suggestion
        If ``data`` is Series then is ignored.
```

taranarmo committed:
make backticks double

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

mroeschke commented:
Thanks @taranarmo

jorisvandenbossche commented:
I was going to comment that I find this a bit surprising behaviour, and so whether we shouldn't rather consider this as a bug instead of documenting it. 

But from a quick check, it seems that the `dtype` argument in fact is _not_ ignored, at least not in all cases (https://github.com/pandas-dev/pandas/issues/59060 clearly shows there are also cases where it is indeed ignored):

```
In [18]: pd.Series(pd.Series([1.0], dtype=float), dtype=int)
Out[18]: 
0    1
dtype: int64
```

Here it does cast the float series to an int64 dtype, so the `dtype` keyword was not ignored.

Will also comment on the issue, but so it seems the simple "If ``data`` is Series then is ignored." message added in this PR is not fully correct.

jbrockmendel commented:
IIUC I agree with Joris. The solution is not to change the doc, but to make the behavior consistent. The dtype kwarg should never be silently ignored.
